karapace/schema_reader.py

```diff
@@ -250,109 +262,131 @@ def handle_messages(self):
                 self.log.info("Read new record: key: %r, value: %r, offset: %r", key, value, msg.offset)
                 self.handle_msg(key, value)
                 self.offset = msg.offset
-                self.log.info("Handled message, current offset: %r", self.offset)
+                self.log.info(
+                    "Handled message, current offset: %r, ready: %r, add_offsets: %r",
+                    self.offset,
+                    self.ready,
+                    add_offsets,
+                )
                 if self.ready and add_offsets:
                     self.offset_watcher.offset_seen(self.offset)
 
-    def handle_msg(self, key: dict, value: dict):
-        if key["keytype"] == "CONFIG":
-            if "subject" in key and key["subject"] is not None:
-                if not value:
-                    self.log.info("Deleting compatibility config completely for subject: %r", key["subject"])
-                    self.subjects[key["subject"]].pop("compatibility", None)
-                    return
-                self.log.info(
-                    "Setting subject: %r config to: %r, value: %r", key["subject"], value["compatibilityLevel"], value
-                )
-                if not key["subject"] in self.subjects:
-                    self.log.info("Adding first version of subject: %r with no schemas", key["subject"])
-                    self.subjects[key["subject"]] = {"schemas": {}}
-                subject_data = self.subjects.get(key["subject"])
-                subject_data["compatibility"] = value["compatibilityLevel"]
-            else:
-                self.log.info("Setting global config to: %r, value: %r", value["compatibilityLevel"], value)
-                self.config["compatibility"] = value["compatibilityLevel"]
-        elif key["keytype"] == "SCHEMA":
+    def _handle_msg_config(self, key: dict, value: Optional[dict]) -> None:
+        subject = key.get("subject")
+        if subject is not None:
+            if subject not in self.subjects:
```

Comment:
This is the fix for the bug mentioned on the side:

https://github.com/aiven/karapace/pull/375/files#r839817686

I just moved the initialization up a bit, so that it is always done.